domain/src/main/java/mihon/domain/chapter/interactor/FilterChaptersForDownload.kt

```diff
@@ -0,0 +1,73 @@
+package mihon.domain.chapter.interactor
+
+import tachiyomi.domain.category.interactor.GetCategories
+import tachiyomi.domain.chapter.interactor.GetChaptersByMangaId
+import tachiyomi.domain.chapter.model.Chapter
+import tachiyomi.domain.download.service.DownloadPreferences
+import tachiyomi.domain.manga.model.Manga
+
+/**
+ * Interactor responsible for determining which chapters of a manga should be downloaded.
+ *
+ * @property getChaptersByMangaId Interactor for retrieving chapters by manga ID.
+ * @property downloadPreferences User preferences related to chapter downloads.
+ * @property getCategories Interactor for retrieving categories associated with a manga.
+ */
+class FilterChaptersForDownload(
+    private val getChaptersByMangaId: GetChaptersByMangaId,
+    private val downloadPreferences: DownloadPreferences,
+    private val getCategories: GetCategories,
+) {
+
+    /**
+     * Determines which chapters of a manga should be downloaded based on user preferences.
+     *
+     * @param manga The manga for which chapters may be downloaded.
+     * @param newChapters The list of new chapters available for the manga.
+     * @return A list of chapters that should be downloaded
+     */
+    suspend fun await(manga: Manga, newChapters: List<Chapter>): List<Chapter> {
+        if (!manga.shouldDownloadNewChapters()) {
+            return emptyList()
+        }
+
+        if (!downloadPreferences.downloadNewUnreadChaptersOnly().get()) return newChapters
+
+        val readChapterNumbers = getChaptersByMangaId.await(manga.id)
+            .filter { it.read && it.isRecognizedNumber }
+            .map { it.chapterNumber }
```

Comment:
`Set` is better in this context
```suggestion
        val readChapterNumbers = buildSet {
            getChaptersByMangaId.await(manga.id).forEach {
        	    if (it.read && it.isRecognizedNumber) add(it.chapterNumber)
            }
        }
```